# This file holds parameters for mola::LidarInertialOdometry,
# for use either programatically calling initilize(), or from a MOLA system
# launch file. See "launch/*" examples.

params:
  # These sensor labels will be handled as LIDAR observations:
  lidar_sensor_labels: ['lidar', 'Velodyne1_SCAN','pandar','livox','os_sensor','KINECT']

  # Optionally, drop lidar data too close in time:
  min_time_between_scans: 0 # 0.2  # [seconds]

  # LIDAR observations are, first, loaded using a generator:
  observations_generator: $include{./observations_generator.yaml}
  
  # then, optionally, filtered before being registered with ICP
  # against the local map:
  observations_filter: $include{./observations_filter_pipeline.yaml}

  # These sensor labels will be handled as IMU observations:
  imu_sensor_label: 'imu'
  
  # How often to update the local map model:
  min_dist_xyz_between_keyframes: 0.1  # [m]
  min_rotation_between_keyframes: 15.0  # [deg]
  
  icp_settings_with_vel: $include{./icp_odom_params.yaml}
  icp_settings_without_vel: $include{./icp_odom_params.yaml}
  #icp_settings_without_vel: $include{./icp_odom_params_no_motion_model.yaml}

  # Minimum ICP quality to insert it into the map:
  min_icp_goodness: 0.30

  adaptive_threshold:
    enabled: true
    initial_sigma: 1.0
    min_motion: 0.05

  # Local map updates:
  # - First time: Using the mp2p_icp pipeline generator:
  localmap_generator: $include{./localmap_generator.yaml}

  # - Rest of observations: use these *pointcloud* layers of the observation 
  #   metric_map_t and *insert* them into the local map:
  observation_layers_to_merge_local_map:
   - {from_obs: 'decimated_for_map_smaller_curvature', into_map: 'localmap_small_curvature'}
   - {from_obs: 'decimated_for_map_large_curvature', into_map: 'localmap_large_curvature'}
  
  # If enabled, a map will be stored in RAM and (if using the CLI) stored
  # to a ".simplemap" file for later use for localization, etc.
  simplemap.generate: false # Can be overriden with CLI flag --output-simplemap
  simplemap.min_dist_xyz_between_keyframes: 0.05   # m
  simplemap.min_rotation_between_keyframes: 3.0  # deg

  # To profile the internal steps of the ICP implementation:
  icp_profiler_enabled: true

